### PR TITLE
Fix tests broken from abstracting the database

### PR DESCRIPTION
**Pull request by sohkai**
Based on merging #895 and #896 into #788.

All tests are a go with Rethinkdb 🚨  (some tests are still coupled to Rethinkdb, but this will be done in the future).

### CODE REVIEW

File: tests/test_core.py [hunk 1]
```diff
@@ -31,9 +32,7 @@ def test_bigchain_class_default_initialization(config):
     from bigchaindb.core import Bigchain
     from bigchaindb.consensus import BaseConsensusRules
     bigchain = Bigchain()
-    assert bigchain.host == config['database']['host']
-    assert bigchain.port == config['database']['port']
-    assert bigchain.dbname == config['database']['name']
+    assert bigchain.connection
```
Comment (sohkai) [inline tests/test_core.py]: Not really sure what to assert anymore; is just checking that `bigchain.connection` enough?

File: tests/test_commands.py [hunk 2]
```diff
@@ -225,10 +215,33 @@ def test_bigchain_run_init_when_db_exists(mock_db_init_with_existing_db):
     run_init(args)
 
 
-def test_drop_existing_db(mock_rethink_db_drop):
+@patch('bigchaindb.backend.schema.drop_database')
+def test_drop_db_when_assumed_yes(mock_db_drop):
```
Comment (sohkai) [inline tests/test_commands.py]: These drop db interactive tests were (more or less) previously done in the `db/test_utils.py` file that was moved to `db/rethinkdb/test_schema.py`.

File: tests/test_core.py [hunk 3]
```diff
@@ -30,10 +31,12 @@ def config(request, monkeypatch):
 def test_bigchain_class_default_initialization(config):
     from bigchaindb.core import Bigchain
     from bigchaindb.consensus import BaseConsensusRules
+    from bigchaindb.backend.connection import Connection
     bigchain = Bigchain()
-    assert bigchain.host == config['database']['host']
-    assert bigchain.port == config['database']['port']
-    assert bigchain.dbname == config['database']['name']
+    assert isinstance(bigchain.connection, Connection)
```
Comment (sohkai) [inline tests/test_core.py]: Wasn't sure if we necessarily wanted to do this check, but I'm k with it.

Comment (vrde) [inline tests/test_core.py]: I wasn't 100% sure as well, but I think it's OK for now. From my experience `asset something` is too loose. I prefer to have a stricter tests that will eventually fail, so we can fix it again in the proper way afterwards.

**Pull request merged**